client/my-sites/backup/rewind-flow/granular-restore.tsx

```diff
@@ -90,6 +101,86 @@ const BackupGranularRestoreFlow: FunctionComponent< Props > = ( {
 
 	const goBackUrl = backupContentsPath( siteSlug as string, rewindId );
 
+	const renderThemes = () => {
+		const themes = browserSelectedList.filter( ( item ) => item.type === 'theme' );
+		if ( themes.length === 0 ) {
+			return null;
+		}
+		return (
+			<>
+				<h4 className="rewind-flow__cta">{ translate( 'WordPress themes' ) }</h4>
+				<ul className="rewind-flow__files">
+					{ themes.map( ( item ) => (
+						<li key={ item.path }>{ item.path }</li>
+					) ) }
+				</ul>
+			</>
+		);
+	};
+
+	const renderPlugins = () => {
+		const plugins = browserSelectedList.filter( ( item ) => item.type === 'plugin' );
+		if ( plugins.length === 0 ) {
+			return null;
+		}
+		return (
+			<>
+				<h4 className="rewind-flow__cta">{ translate( 'WordPress plugins' ) }</h4>
+				<ul className="rewind-flow__files">
+					{ plugins.map( ( item ) => (
+						<li key={ item.path }>{ item.path }</li>
+					) ) }
+				</ul>
+			</>
+		);
+	};
+
+	const renderTables = () => {
+		const tables = browserSelectedList.filter( ( item ) => item.type === 'table' );
+		if ( tables.length === 0 ) {
+			return null;
+		}
+		return (
+			<>
+				<h4 className="rewind-flow__cta">{ translate( 'Site database' ) }</h4>
+				<ul className="rewind-flow__files">
+					{ tables.map( ( item ) => (
+						<li key={ item.path }>{ item.path }</li>
+					) ) }
+				</ul>
+			</>
+		);
+	};
+
+	const renderExpandIcon = () => {
+		return <Icon icon={ showAllFiles ? chevronDown : chevronRight } />;
+	};
+
+	const renderFiles = () => {
```

Comment:
If you try to restore only 1 file (or maybe multiple files, but no plugin/theme/tables) it will show all files inside the expandable div:

<img width="723" alt="CleanShot 2023-09-04 at 19 18 48@2x" src="https://github.com/Automattic/wp-calypso/assets/1488641/8dd0636a-a5b5-4f6a-ba5c-bad22fe0c933">

Maybe we could validate if there's any plugin, theme or tables and if there's none, we could list the files the same way we do right now. Any thoughts?

![image](https://github.com/Automattic/wp-calypso/assets/1488641/867bb843-f269-4418-95df-15963f1d0953)

Reply:
> Maybe we could validate if there's any plugin, theme or tables and if there's none, we could list the files the same way we do right now. Any thoughts?

That seems good, or we can set a number of files to display, and then x more or something expandable.